third_party/machinarium/sources/clock.c

```diff
@@ -9,14 +9,15 @@
 #include <machinarium_private.h>
 
 static int
-mm_clock_cmp(const void *a_ptr, const void *b_ptr)
+mm_clock_cmp(mm_timer_t *a, mm_timer_t *b)
 {
-	mm_timer_t *a = *(mm_timer_t**)a_ptr;
-	mm_timer_t *b = *(mm_timer_t**)b_ptr;
 	if (a->timeout == b->timeout) {
+		if (a->seq == b->seq) {
+			assert(a == b);
+			return 0;
+		}
 		return (a->seq > b->seq) ? 1 : -1;
-	} else
```

Comment:
As far as i understand we use this in following form in odyssey:)

if {
} else
if